tests/Resources/ti.ui.animation.test.js

```diff
@@ -0,0 +1,420 @@
+				instance.transform = Ti.UI.createMatrix2D({
+					scale: 2
+				});
+				should(instance).have.a.property('transform').which.is.an.Object();
+				const matrixApiName = Ti.Platform.osname === 'android' ? 'Ti.UI.Matrix2D' : 'Ti.UI.2DMatrix';
+				should(instance.transform.apiName).eql(matrixApiName);
+			});
+		});
+
+		describe.ios('.transition', () => {
+			it('is a Number', () => {
+				instance.transition = Ti.UI.iOS.AnimationStyle.CROSS_DISSOLVE;
+				should(instance).have.a.property('transition').which.is.a.Number();
+			});
+
+			it('is one of Ti.UI.iOS.AnimationStyle.CURL_*', () => {
+				instance.transition = Ti.UI.iOS.AnimationStyle.FLIP_FROM_TOP;
+				should([
+					Ti.UI.iOS.AnimationStyle.CURL_DOWN,
+					Ti.UI.iOS.AnimationStyle.CURL_UP,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_LEFT,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_RIGHT,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_TOP,
+					Ti.UI.iOS.AnimationStyle.FLIP_FROM_BOTTOM,
+					Ti.UI.iOS.AnimationStyle.CROSS_DISSOLVE,
+					Ti.UI.iOS.AnimationStyle.NONE,
+				]).containEql(instance.transition);
+			});
+		});
+
+		describe.ios('.view', () => {
+			it('is a Titanium.UI.View', () => {
+				instance.view = Ti.UI.createView();
+				should(instance).have.a.property('view').which.is.an.Object();
+				should(instance.view).have.a.property('apiName').which.is.eql('Ti.UI.View');
+			});
+		});
+
+		describe.ios('.visible', () => {
+			it('is a Boolean', () => {
+				instance.visible = false;
+				should(instance).have.a.property('visible').which.is.a.Boolean();
+			});
+		});
+
+		describe('.width', () => {
+			it('is a Number', () => {
+				instance.width = 256;
+				should(instance).have.a.property('width').which.is.a.Number();
+			});
+		});
+
+		describe.ios('.zIndex', () => {
+			it('is a Number', () => {
+				instance.zIndex = 2;
+				should(instance).have.a.property('zIndex').which.is.a.Number();
+			});
+		});
+	});
+
+	describe('examples', () => {
+		let win;
+		let container;
+
+		function once(target, name, listener) {
+			target.addEventListener(name, e => {
+				target.removeEventListener(name, listener);
+				listener(e);
+			});
+		}
+
+		afterEach(done => {
+			if (win && !win.closed) {
+				win.addEventListener('close', function listener () {
+					win.removeEventListener('close', listener);
+					container = null;
+					win = null;
+					done();
+				});
+				console.log('win.close()');
+				win.close();
+			} else {
+				container = null;
+				win = null;
+				done();
+			}
+		});
+
+		beforeEach(() => {
+			win = Ti.UI.createWindow();
+			container = Ti.UI.createView({
+				width: 300,
+				height: 300,
+				backgroundColor: 'white'
+			});
+			win.add(container);
+		});
+
+		it('Simple Animation Applied to a View', done => {
+			var view = Titanium.UI.createView({
+				backgroundColor: 'red'
+			});
+			container.add(view);
+			once(win, 'postlayout', () => {
+				should(container).matchImage('snapshots/animation_simple_start.png');
```

Comment:
I wrap the async/callback aerations in try/catch blocks with the catch calling `finish(err);`, otherwise we don't behave as expected and the test times out rather than surfaces the Error that occurred async. So basically anything in any event callback/timer needs to be wrapped din `try`/`catch` to manually call `finish(err)` to bubble up the true error. (I assume that's a behavioral difference/bug we have versus Node!)